From d90af2d6b09bc5197a39d1e1d09baff2aafe9a2a Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Tue, 3 Jun 2014 13:43:04 +1000
Subject: [PATCH 02/16] Don't attempt to issue DB query if there's nothing to
 do

Prevent harmless SQL statement error messages to be logged when the position map is empty

Regression introduced in b4be8e24f9fab5b6a71f62df8e2c0403bedf4805

Fixes #12161
---
 mythtv/libs/libmyth/programinfo.cpp |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mythtv/libs/libmyth/programinfo.cpp b/mythtv/libs/libmyth/programinfo.cpp
index 9203f1a..e3c8737 100644
--- a/mythtv/libs/libmyth/programinfo.cpp
+++ b/mythtv/libs/libmyth/programinfo.cpp
@@ -3710,6 +3710,9 @@ void ProgramInfo::SavePositionMap(
     if (!query.exec())
         MythDB::DBError("position map clear", query);
 
+    if (posMap.isEmpty())
+        return;
+
     // Use the multi-value insert syntax to reduce database I/O
     QStringList q("INSERT INTO ");
     QString qfields;
@@ -3765,6 +3768,9 @@ void ProgramInfo::SavePositionMap(
 void ProgramInfo::SavePositionMapDelta(
     frm_pos_map_t &posMap, MarkTypes type) const
 {
+    if (posMap.isEmpty())
+        return;
+
     if (positionMapDBReplacement)
     {
         QMutexLocker locker(positionMapDBReplacement->lock);
-- 
1.7.10.2

